src/wallet/rpc/wallet.cpp

```diff
@@ -701,6 +701,40 @@ RPCHelpMan simulaterawtransaction()
     };
 }
 
+static RPCHelpMan migratewallet()
+{
+    return RPCHelpMan{"migratewallet",
+        "\nEXPERIMENTAL warning: This call may not work as expected and may be changed in future releases\n"
+        "\nMigrate the wallet to a descriptor wallet.\n"
+        "A new wallet backup will need to be made.\n"
+        "\nThe migration process will create a backup of the wallet before migrating. This backup\n"
+        "file will be named <wallet name>-<timestamp>.legacy.bak and can be found in the directory\n"
+        "for this wallet. In the event of an incorrect migration, the backup can be restored using restorewallet." +
+        HELP_REQUIRING_PASSPHRASE,
+        {},
+        RPCResult{RPCResult::Type::NONE, "", ""},
+        RPCExamples{
+            HelpExampleCli("migratewallet", "")
+            + HelpExampleRpc("migratewallet", "")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    std::shared_ptr<CWallet> wallet = GetWalletForJSONRPCRequest(request);
+    if (!wallet) return NullUniValue;
+
+    EnsureWalletIsUnlocked(*wallet);
+
+    WalletContext& context = EnsureWalletContext(request.context);
+
+    bilingual_str error;
+    if (!MigrateLegacyToDescriptor(std::move(wallet), context, error)) {
+        throw JSONRPCError(RPC_WALLET_ERROR, error.original);
+    }
+    return NullUniValue;
```

Comment:
Maybe it can return a success message and others indicating the backup, watchonly and multisig file path?

Reply:
Done